WcaOnRails/app/controllers/competitions_controller.rb

```diff
@@ -69,6 +62,15 @@ def index
       @competitions = @competitions.select { |competition| competition.contains?(params[:search]) }
     end
 
+    unless params[:event_ids].empty?
+      @competitions = @competitions.select do |competition|
+        (competition.eventSpecs.split & params[:event_ids]).any?
+      end.sort_by do |competition|
+        -(competition.eventSpecs.split & params[:event_ids]).count
```

Comment:
Haha, It's actually what I've done today, but we end up with the same as before (recomputing the number), if you consider this option to be good enough, I will do this!